### PR TITLE
Move admin handler to their own dir

### PR DESCRIPTION
**Pull request by jwolski**
This might look like a lot of shuffle. But there's 0 functionality change (famous last words). Though completely unrelated, it's in preparation for landing damp scoring. The reason this change exists is because a new admin handler is needed in flap damping to apply arbitrary membership updates to a node. That handler is still to come (in another PR), but I think this re-org makes the admin API more intuitive.

In short, this places all admin API endpoint handlers into a `server/admin` directory. And the `index.js` within defines both the URL/endpoint and the handler function for each.

@danielheller @CorgiMan @benfleis @Raynos

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "pre-commit": "^0.0.9",
     "tape": "^3.0.3",
     "tape-cluster": "2.1.0",
-    "tchannel": "^2.5.1",
+    "tchannel": "2.5.1",
```
Comment (jwolski) [inline package.json]: This is a dev dependency of tchannel. A recent tchannel update broke tests. This is in an effort to pass this test run and potentially fix in another PR. You can ignore this change for now.

File: server/admin/debug-clear.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
```
Comment (jwolski) [inline server/admin/debug-clear.js]: NOTE FOR REVIEWER: same functionality as before. just moved out of `server/index.js` and into its own file.

File: server/admin/debug-set.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
```
Comment (jwolski) [inline server/admin/debug-set.js]: same functionality as before. just moved out of `server/index.js` and into its own file.

File: server/admin/join.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
```
Comment (jwolski) [inline server/admin/join.js]: same functionality as before. just moved out of `server/index.js` and into its own file.

File: server/admin/leave.js [hunk 5]
```diff
@@ -19,7 +19,7 @@
 // THE SOFTWARE.
 'use strict';
 
-var errors = require('../lib/errors.js');
+var errors = require('../../lib/errors.js');
```
Comment (jwolski) [inline server/admin/leave.js]: same functionality as before. just moved out of `server/index.js` and into its own file.

File: server/admin/lookup.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
```
Comment (jwolski) [inline server/admin/lookup.js]: same functionality as before. just moved out of `server/index.js` and into its own file.

File: server/admin/reload.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
```
Comment (jwolski) [inline server/admin/reload.js]: same functionality as before. just moved out of `server/index.js` and into its own file.

File: server/admin/stats.js [hunk 8]
```diff
@@ -19,8 +19,8 @@
 // THE SOFTWARE.
 'use strict';
 
-module.exports = function handleAdminLookup(opts, callback) {
-    callback(null, null, {
-        dest: opts.ringpop.lookup(opts.key)
-    });
+module.exports = function createHandler(ringpop) {
```
Comment (jwolski) [inline server/admin/stats.js]: same functionality as before. just renamed file

File: server/admin/tick.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
```
Comment (jwolski) [inline server/admin/tick.js]: same functionality as before. just moved out of `server/index.js` and into its own file.

**Comment by benfleis** [comment]
@jwolski 

Overall questions:
 - same functionality ?= same code, just shifted (It could mean reimplemented w/ same in/out, in which case I should be reading the actual content as well.)
 - all of the createHandle and handle lambda fns lost their name specificity. the filename will show in the trace, but it might still be nice to have the redundant fn names in their. Thoughts?

Assume above `functionality === code`, then lgtm. If not, then I'll give a better read pass.

**Comment by jwolski** [comment]
good point re: fn names. I will fix.

**Comment by jwolski** [comment]
sorry. and yes, it is the "same code".

**Comment by jwolski** [comment]
actually, there was a slight change in endpoint registration logic (to make it generic for both admin + protocol endpoints). you can take a look at that here: https://github.com/uber/ringpop/pull/145/files#diff-0861d6d6b50d7d695344bf2d86d6e5e6R42
